snuba/query/processors/tagsmap.py

```diff
@@ -136,6 +145,29 @@ def process_query(self, query: Query, request_settings: RequestSettings) -> None
             if not apply_optimization:
                 return
 
+        # Do not use flattened tags if tags are being unpacked anyway. In that case
+        # using flattened tags only implies loading an additional column thus making
+        # the query heavier and slower
+        if query.get_arrayjoin_from_ast() and self.__find_tags(
```

Comment:
Sorry, had to support only the new AST. Doing this in the old model would have been a huge pain.
In practice we are already parsing the query into the new AST, so it is properly formatted when we get here, if there is a failure in parsing it would be None and the optimization would not be applied. Correctness does not change so no big deal.

Reply:
I think this is fine.